### PR TITLE
allow to use different consumers for the same messages but from different queues

### PR DESCRIPTION
**Pull request by vitaly-zhuravkov**

### CODE REVIEW

**Comment by VladislavMorozovIDT** [comment]
Could you also refactor ?
image

File: src/SqsPoller/ConsumerResolver.cs [hunk 1]
```diff
@@ -6,28 +6,29 @@
...
 using Microsoft.Extensions.Logging;
 
 namespace SqsPoller
 {
 
     public class ConsumerResolver: IConsumerResolver
     {
-        private readonly IEnumerable<(IConsumer instance, Type messageType, SqsConsumer mapping)> _consumersMapping;
+        private readonly IEnumerable<(Type consumerType, Type messageType, SqsConsumer mapping)> _consumersMapping;
+        private readonly IServiceProvider _serviceProvider;
```
Comment (VladislavMorozovIDT) [inline src/SqsPoller/ConsumerResolver.cs:18]: Looks like this field isn't reusable

Comment (vitaly-zhuravkov) [reply src/SqsPoller/ConsumerResolver.cs:18]: What do you mean?

Comment (VladislavMorozovIDT) [reply src/SqsPoller/ConsumerResolver.cs:18]: Sorry, I was wrong

**Review by VladislavMorozovIDT — APPROVED** [review]

**Review by AlexeyBogdan95 — APPROVED** [review]

**Pull request merged**